src/doc/src/reference/environment-variables.md

```diff
@@ -214,6 +214,7 @@ corresponding environment variable is set to the empty string, `""`.
 * `CARGO_PKG_REPOSITORY` — The repository from the manifest of your package.
 * `CARGO_PKG_LICENSE` — The license from the manifest of your package.
 * `CARGO_PKG_LICENSE_FILE` — The license file from the manifest of your package.
+* `CARGO_PKG_RUST_VERSION` — The Rust version from the manifest of your package.
```

Comment:
The thing going through my head is the question: "what principle governs us exposing the manifest as environment variables?". 

For myself, exposing `CARGO_PKG_RUST_VERSION` feels very arbitrary from a cargo-design perspective.  Why this field and not many of the others?

Reply:
Many of the other fields are already exposted. Like the ``version`` field is even exposed through 4 or 5 env vars. Most package metadata is exported. I think this makes it more consistent.

Reply:
Also see https://github.com/rust-lang/cargo/pull/8758 where CARGO_PRIMARY_PACKAGE was added "primarily for Clippy".

Reply:
> Many of the other fields are already exposed. Like the version field is even exposed through 4 or 5 env vars. Most package metadata is exposed. I think this makes it more consistent.

Except for CARGO_PRIMARY_PACKAGE , those all look targeted at the code being built, either so a crate can access its own metadata or build.rs / test purposes.

> Also see https://github.com/rust-lang/cargo/pull/8758 where CARGO_PRIMARY_PACKAGE was added "primarily for Clippy".

Thanks for pointing out precedence on this.  I am still interested in seeing if there is an answer to my question or if there is interest in creating an answer to my question. Until that time, I am not comfortable merging this though maybe others of the cargo team would be.

Reply:
I'm good with waiting on other team members. Just wanted to bring this up through a PR again, because my question some time ago on Zulip didn't get any replies 😅

Reply:
We discussed this in the cargo team meeting and we have no guiding principle on what env variables should exist and are intending to be responsive to what different teams needs are.  I'm fine merging this after the documentation gets improved.